### PR TITLE
Adding missing workflow_step

### PR DESCRIPTION
**Pull request by JerRMartin**
### Motivation

Which issue does this fix? N/A

If no issue exists, what is the fix or new feature? Were there any reasons to fix/implement things that are not obvious?
- adding in `workflow_step` field as part of `System` class
    - Documentation [here](https://kontent.ai/learn/reference/delivery-api/#tag/Linked-content-and-components) lists `workflow_step` in the response, but Java SDK has no field to access this data on an initialized System Object. 
### Checklist

- [X] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [X] Tests are passing
- [ ] Docs have been updated (if applicable)
- [X] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

If manual testing is required, what are the steps?

In the Main.java file, edit the `DeliveryOptions` on line 13 to target a test Kontent Project and to look like this:
```
DeliveryOptions options = DeliveryOptions.builder()
                .projectId("<YOUR-PROJECT-ID>")
                .retryAttempts(2)
                .waitForLoadingNewContent(true)
                .previewApiKey("<YOUR-PREVIEW-API-KEY>")
                .usePreviewApi(true)
                .build();
```

#### Step 2:
Rename `client` to `client2` on line 23, then remove the `.getName()` from line 25, and reference an item in the test Kontent Project like so: 
```
        client2.getItem("<KONTENT-ITEM-CODENAME>")
                .thenAccept(item ->
                        System.out.println(item.getItem().getSystem()));
```

#### Expected Results: 
When targeting a test project with an unpublished element you should see a System object printed out displaying the `workflow_step` as one of the fields like so: 
```
System(id=<SOME-ID>, name=<SOME-NAME>, codename=<SOME-CODENAME>, language=default, type=<SOME-TYPE>, collection=default, sitemapLocations=[], lastModified=<SOME-TIMESTAMP>, workflowStep=draft)

```

### CODE REVIEW

**Comment by Simply007** [comment]
Hello @JerRMartin,

thanks for the PR. The change itself seems fine. But the tests are not fully updated and I would love to have this information also in the readme.

According to the changes, it is conceptually the same as in including collection to the SDK:
* https://github.com/Kentico/kontent-java-packages/pull/119/files

So please adjust all JSON mock files, extend the binding test and filtering test and extend readme and the PR should be ready to go!

> The `workflow_step` is only being set for all content items - for rich text components it is not: https://kontent.ai/learn/reference/delivery-api/#section/Content-item-object, but I haven't found any components in testing data. It would be great to extend the `SampleContentItem.json`'s description element with one component without having workflow steps and test if it is correctly being set. I would keep this one optional and we can submit a separate issue if not included in this PR.

Once these are in place, I can publish a snapshot to the maven central.

**Comment by JerRMartin** [comment]
> Thank you for your edits!
> 
> Could you reformat `README.md` to keep its shape and just leave the Workflow step part addition?

I'm sorry, I don't know that I understood what you meant by this. Would you mind rewording it?

**Comment by Simply007** [comment]
> > Thank you for your edits!
> > Could you reformat `README.md` to keep its shape and just leave the Workflow step part addition?
> 
> I'm sorry, I don't know that I understood what you meant by this. Would you mind rewording it?

Sure, In commit https://github.com/Kentico/kontent-java-packages/pull/139/commits/0567fb2ebddb5ef12b10ee38b2d80079c53042f2 you adjusted the readme [with these changes](https://github.com/Kentico/kontent-java-packages/pull/139/files#diff-028248a9ba651e4821242aca3450130c903fb9c452060199ed22c36e3952b7e2R152-R175)  which are completely correct.

image

But there are other changes that are changing the format of the README, like this one:
image
So I would like to have these other changes reverted.

**Comment by JerRMartin** [comment]
> > > Thank you for your edits!
> > > Could you reformat `README.md` to keep its shape and just leave the Workflow step part addition?
> > 
> > 
> > I'm sorry, I don't know that I understood what you meant by this. Would you mind rewording it?
> 
> Sure, In commit [0567fb2](https://github.com/Kentico/kontent-java-packages/commit/0567fb2ebddb5ef12b10ee38b2d80079c53042f2) you adjusted the readme [with these changes](https://github.com/Kentico/kontent-java-packages/pull/139/files#diff-028248a9ba651e4821242aca3450130c903fb9c452060199ed22c36e3952b7e2R152-R175) which are completely correct.
> 
> image
> 
> But there are other changes that are changing the format of the README, like this one: image So I would like to have these other changes reverted.

Sorry about that, it seems like the Markdown editor I used auto formatted the file without me noticing. Good Catch!

File: kontent-delivery/README.md [hunk 1]
```diff
@@ -172,27 +175,30 @@ When retrieving a list of content items, you get an instance of the `ContentItem
...
+// Retrieves codename of an article content item  
+articleItem.getSystem().getCodename()  
+  
+// Retrieves codename of the collection of an article content item  
+articleItem.getSystem().getCollection()  
+  
+// Retrieves codename of the content type of an article content item  
+articleItem.getSystem().getType()  
+
+// Retrieves the Workflow Step of an article content item
```
Comment (Simply007) [inline kontent-delivery/README.md:199]: // Retrieves codename of the workflow step of an article content item

Comment (Simply007) [inline kontent-delivery/README.md:199]:
```suggestion
// Retrieves codename of the workflow step of an article content item
```

File: kontent-delivery/README.md [hunk 2]
```diff
@@ -308,7 +314,8 @@ We would like to express our thanks to the following people who contributed and
 - [Adam J. Weigold](https://github.com/aweigold)
 - [Tommaso Garuglieri](https://github.com/GaruGaru)
 - [Gabriel Cunha](https://github.com/cunhazera)
+- [Jeremy Martin](https://github.com/jerrmartin)
 
 Would you like to become a hero too? Pick an [issue](https://github.com/Kentico/kontent-delivery-sdk-java/issues) and send us a pull request!
 
-![Analytics](https://kentico-ga-beacon.azurewebsites.net/api/UA-69014260-4/Kentico/kontent-java-packages/kontent-delivery?pixel)
+![Analytics](https://kentico-ga-beacon.azurewebsites.net/api/UA-69014260-4/Kentico/kontent-java-packages/kontent-delivery?pixel)
```
Comment (Simply007) [inline kontent-delivery/README.md:321]:
```suggestion
```

**Review by Simply007 — APPROVED** [review]
Thank you @JerRMartin,

changes look OK. I have adjusted two minor things.

Later today, I will merge the PR and release the snapshot for testing. If everything works OK, I can promote shapshot to a new minor version!

**Comment by Simply007** [comment]
Hello @JerRMartin,

I have merged the PR and released [4.6.0-SNAPSHOT](https://github.com/Kentico/kontent-java-packages/releases/tag/4.6.0-SNAPSHOT). 

Could you test out, if everything works as expected on your particular project (the guide is [here](https://github.com/Kentico/kontent-java-packages/wiki/Release-process#snapshot-release---xyz-snapshot-vs-xyz))?

Once confirmed, I can promote the snapshot to the production version.

**Comment by JerRMartin** [comment]
> Hello @JerRMartin,
> 
> I have merged the PR and released [4.6.0-SNAPSHOT](https://github.com/Kentico/kontent-java-packages/releases/tag/4.6.0-SNAPSHOT).
> 
> Could you test out, if everything works as expected on your particular project (the guide is [here](https://github.com/Kentico/kontent-java-packages/wiki/Release-process#snapshot-release---xyz-snapshot-vs-xyz))?
> 
> Once confirmed, I can promote the snapshot to the production version.

Looks, Great! I'd say its ready to be promoted!
Screen Shot 2022-06-20 at 12 47 47 PM

**Comment by Simply007** [comment]
Version 4.6.0 is out!

https://repo1.maven.org/maven2/com/github/kentico/kontent-delivery/

**Pull request merged**